### PR TITLE
Fix SliceChannel Type inference

### PR DESCRIPTION
**Pull request by anirudh2290**
## Description ##
Fix SliceChannel Type Inference. Do forward and backward inference for slice channel with ElemwiseAttr logic. Remove exception thrown from infer_type logic in slice channel.
Adds some gluoncv models to AMP examples.

@ptrendx 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/slice_channel-inl.h [hunk 1]
```diff
@@ -176,16 +177,22 @@ class SliceChannelProp : public OperatorProperty {
   bool InferType(std::vector<int> *in_type,
                  std::vector<int> *out_type,
                  std::vector<int> *aux_type) const override {
+    /*
```
Comment (ChaiBapchya) [inline src/operator/slice_channel-inl.h]: If this is to be removed, why is it commented? Or is this temporary?

File: tests/python/gpu/test_contrib_amp.py [hunk 2]
```diff
@@ -475,6 +475,15 @@ def test_fp16_casting():
...
     assert exe.arg_arrays[0].dtype == np.float16
 
+    # Check for symbol which has slice channel
+    data = mx.sym.var("data")
+    data2 = mx.sym.var("data2")
+    data._set_attr(__dtype__="-1")
+    data2._set_attr(__dtype__="-1")
+    concat_res = mx.sym.concat(data, data2)
+    out = mx.sym.split(concat_res, axis=1, num_outputs=2)
+    final_res = amp.convert_symbol(out)
```
Comment (ChaiBapchya) [inline tests/python/gpu/test_contrib_amp.py]: Any checks or assertions needed?

File: src/operator/slice_channel-inl.h (hunk 1 above)
Comment (anirudh2290) [reply src/operator/slice_channel-inl.h]: yes, it should have been removed. Good catch, fixed.

File: tests/python/gpu/test_contrib_amp.py (hunk 2 above)
Comment (anirudh2290) [reply tests/python/gpu/test_contrib_amp.py]: no, this was earlier failing in the convert_symbol call before this change. Just need to check if convert_symbol completes successfully.

**Comment by ptrendx** [comment]
I don't know enough to comment on the changes to the example etc., but the main fix from this PR (InferType in SliceChannel) looks good. @anirudh2290 Searching for `First input must have specified type` gives many other hits, do you intend to look into those as well?

**Comment by anirudh2290** [comment]
@ptrendx how did you come up with that regex ?

**Comment by anirudh2290** [comment]
@ptrendx good point about the other places where error is raised. The scope of the PR increased :) . Can we start with this PR first and I will follow up with another PR/PRs for fixes in other ops. Sorry, little busy with other deliverables too and cannot do it immediately.

For the examples, I have tested running the gluon models for conversion and dummy inference in a loop.

**Comment by ptrendx** [comment]
Sure, having this PR fix just SliceChannel is totally fine.

**Pull request merged**